test/e2e/app/app.go

```diff
@@ -568,30 +568,48 @@ func parseTx(tx []byte) (string, string, error) {
 	return string(parts[0]), string(parts[1]), nil
 }
 
-func (app *Application) verifyAndSum(areExtensionsEnabled bool, currentHeight int64, extCommit *abci.ExtendedCommitInfo, callsite string) (int64, error) {
+func (app *Application) verifyAndSum(
+	areExtensionsEnabled bool,
+	currentHeight int64,
+	extCommit *abci.ExtendedCommitInfo,
+	callsite string,
+) (int64, error) {
 	var sum int64
 	var extCount int
 	for _, vote := range extCommit.Votes {
-		if !vote.SignedLastBlock {
-			continue
+		if vote.BlockIdFlag == cmtproto.BlockIDFlagUnknown || vote.BlockIdFlag > cmtproto.BlockIDFlagNil {
+			return 0, fmt.Errorf("vote with bad blockID flag value at height %d; blockID flag %d", currentHeight, vote.BlockIdFlag)
 		}
-		if !areExtensionsEnabled {
+		if vote.BlockIdFlag == cmtproto.BlockIDFlagAbsent || vote.BlockIdFlag == cmtproto.BlockIDFlagNil {
 			if len(vote.VoteExtension) != 0 {
-				return 0, fmt.Errorf("non-empty vote extension at height %d, which has extensions disabled", currentHeight)
+				return 0, fmt.Errorf("non-empty vote extension at height %d, for a vote with blockID  flag %d",
```

Comment:
Essentially, if this validator had voted correctly, and there is an extension, the flag had to have been Commit?

Reply:
Exactly:
* BlockIDFlagAbsent means it didn't precommit so, how come we got a vote extension?
* BlockIDFlagNil mean it precommitted for `nil`, but `nil` precommits don't have vote extensions

In both cases the info we get is contradictory, so we should reject it.